### PR TITLE
fix: log error in case configuration not found

### PR DESCRIPTION
**Pull request by byte-bandit**
## Changes Proposed:
This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

If your pull request promotes complex changes that require a detailed explanation, please describe them in detail specifying what your solution is and what is it meant to address.

## Issues Addressed:
- See https://github.com/azerothcore/azerothcore-wotlk/issues/17063

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Manually tested by the author.
- [ ] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [x] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. Create a new DB
2. Run dbimport

## Known Issues and TODO List:

None

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by michaeldelago** [comment]
I'm pretty sure the worldserver/authserver print the exceptions error as a part of the Config.cpp file that sources the config, without any extra behavior on their side. 

Can the implementation that works for dbimport remain consistent with the main 2 services?

**Comment by Nyeriah** [comment]
Friendly reminder that this is the standalone database assembler (not part of the core)

File: src/tools/dbimport/Main.cpp [hunk 1]
```diff
@@ -52,6 +52,18 @@ int main(int argc, char** argv)
...
 
+    std::error_code ec;
+    if (!fs::exists(configFile, ec))
+    {
+        if (ec)
+        {
+            printf("failed to read configuration file: '%s'\n", ec.message().c_str());
+        }
+
+        printf("configuration file not found.");
```
[the remark this answers is not in the document]
Comment (Kitzunu) [reply src/tools/dbimport/Main.cpp:63]: Yes initiate logging earlier

**Comment by byte-bandit** [comment]
> I'm pretty sure the worldserver/authserver print the exceptions error as a part of the Config.cpp file that parses/loads the config, without extra behavior in their Main.cpp.
> 
> Can the implementation that works for dbimport remain consistent with the main 2 services?

Good catch, I didn't realise that. Did some quick testing, and you're correct. I will take a closer look at this once I get the time and update the PR.

**Comment by byte-bandit** [comment]
Interestingly, the setup of configuration and logging is completely identical between world/auth server and the dbimport tool. The docker compose deployment differs, but even when using identical settings, dbimport simply won't print out the error.

I will be travelling for the next 10 days or so and won't have access to my main machine, but I'll take another closer look once I'm back.

File: src/tools/dbimport/Main.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (byte-bandit) [reply src/tools/dbimport/Main.cpp]: Removed as no longer needed.

**Comment by byte-bandit** [comment]
So, I finally got back to this and after some more digging found the culprit. Logging is actually not set up and will not work at this early stage. Trying to set it up any earlier also fails due to the missing config.

The reason why the error is correctly printed for `worldserver` and `authserver` is because during config loading, a check is performed against the file name. Here is where the magic happens: https://github.com/azerothcore/azerothcore-wotlk/blob/76d1d02dce2f63348dccb8ac218ec0cc54e6ce8c/src/common/Configuration/Config.cpp#L60

If it's one of the server configuration files, we actually don't use `LOG_ERROR`, but a simple `fmt::Printf`. Adding the configuration file of the `dbimport` changes this behaviour so that we get useful error outputs in case of a missing configuration file :)

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**